### ISSUE TITLE
Much laggier than polymorphicshade/official versions

### ISSUE THREAD

**Issue by fuzxi** [javulticat/NewPipe#3]
### Checklist

- [X] I am able to reproduce the bug with the latest version given here: [CLICK THIS LINK](https://github.com/javulticat/NewPipe/releases/latest).
- [X] I am *not* able to reproduce the bug with the *same version* of [upstream NewPipe](https://github.com/TeamNewPipe/NewPipe/releases/).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/javulticat/NewPipe/issues) or [closed](https://github.com/javulticat/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/javulticat/NewPipe/blob/sponsorblock/.github/CONTRIBUTING.md).

### Affected version

0.26.1

### Steps to reproduce the bug

Hey, thanks for picking up this fork. I made a backup of my database from the polymorphicshade fork, restored it in this version, and noticed that it runs much worse. Scrolling is laggy, videos take several seconds to load and switch between fullscreen/windowed mode, etc. I then restored the same backup into v0.26.1 of the official newpipe build and had none of the same issues.
In their 0.26.0 release [testing issue](https://github.com/TeamNewPipe/NewPipe/issues/10471), some people complained of these issues and [a dev replied](https://github.com/TeamNewPipe/NewPipe/issues/10471#issuecomment-1809155754) that that's to be expected with dev releases. I guess they do something different in the build process for stable vs. dev releases that optimizes the app?

**Comment by fuzxi** [comment]
It looks like this is indeed because debug tools are still included in this build. Long-pressing this build's icon in my launcher shows an option to open the included LeakCanary extension, which is not present in the polymorphicshade/official release builds.

**Comment by javulticat** [comment]
**TL;DR:** Thanks for reporting! I'll be working on this today!

Ah, I knew that releasing debug APKs (which, yes, you are correct that I released a debug APK for `v0.26.1` for expediency's sake, as debug APKs get produced automatically by the CI/CD pipeline we inherit from upstream NewPipe), wasn't going to be a long-term solution (because, at the very least, they take up unnecessary storage space), but I hadn't seen any adverse performance from using it on any of my devices.  But, if the debug APKs are also causing performance issues on certain devices, then I will handle this ASAP.

AFAIK, both upstream and `polymorphicshade` utilize(d) F-Droid's build process: upstream APKs seemed to get built by both their own F-Droid server and the official F-Droid server, while `polymorphicshade`'s were built by IzzyOnDroid's F-Droid server. While I'd love to get this fork published in IzzyOnDroid's repo (and even the official F-Droid repo, if possible), I don't know how realistic that is right now, given how young this fork is at the moment.  Of course, anyone is still more than welcome to [make a request for this project to be added to IzzyOnDroid's repo](https://gitlab.com/IzzyOnDroid/repo/-/issues), if being able to download/update the app though an F-Droid app manager is important to the community.  If anyone does so, I'd be happy to support that effort in any way I can.

However, F-Droid support can/should be tracked/handled in a separate thread.  The GitHub CI/CD pipeline we've inherited from upstream can be configured to automatically build (non-debug) release APKs whenever a new release is published, which will solve the issue independently from anything relating to F-Droid.  Plus, this should make this repo work seamlessly with any app manager capable of installing releases directly from GitHub (e.g., [Obtainium](https://github.com/ImranR98/Obtainium#readme)).

Building release APKs using the repo's own CI/CD pipeline also comes with, IMO, a major intrinsic benefit: it will leave a very obvious "paper trail" in the logs from the CI/CD pipeline runs (found in the repo's [Actions](https://github.com/javulticat/NewPipe/actions)), which will show the exact process by which the release APKs get built directly from the source code.  So, I think this will serve an important role in increasing the transparency of the project, regardless of whether the project ends up in an F-Droid repo at some point. AFAICT, neither the upstream nor `polymorphicshade` repos contain obvious "paper trails" of the build process that is creating the APKs attached to their Releases.

IMHO, it's important in FOSS for the build process to be as transparent as possible - the easier you make it for anyone to be able to see/audit how releases get built, the more it helps to remove the need for trust.  I think it's especially important in cases like this, where a new fork is attempting to carry on work done by a developer who had built trust over years within a community.  I imagine most of that community is likely unfamiliar with any of the other popular FOSS projects I maintain (e.g., [Zappa](https://github.com/zappa/Zappa)), contribute to, and/or consult on, as the main audience of most of the projects I'm involved with is typically other developers.  However, the primary audience here is end-users.  So, while I hope to earn this community's trust, I also intend to eliminate the need for it. 🙃

Anyway, sorry for the long-winded response!  As my first direct interaction with a member of the community, I decided to use the opportunity to also address the community in general to help share a little bit about my FOSS background, my philosophy/thoughts/general musings on this project and FOSS in general.  And a big thanks to everyone for your support! 🙏

**Comment by fuzxi** [comment]
Thanks, I appreciate the explanation! That all sounds perfectly reasonable and I'll be looking forward to the next release.

**Comment by javulticat** [comment]
Hey, just wanted to let you know that you can now get a fresh release APK [here](https://github.com/javulticat/NewPipe/releases/tag/v0.26.1-2).  And, as promised, the build process now gets [logged](https://github.com/javulticat/NewPipe/actions/runs/7408114697/job/20155684402) and is fully automated by GitHub with no humans necessary. 🤖

Hopefully using this version will solve the performance issues you were having!

**Comment by fuzxi** [comment]
Yep, the new version has no performance issues. Thanks for getting that out so quickly!
